pandas/tests/io/test_excel.py

```diff
@@ -1414,133 +1304,114 @@ def test_excel_date_datetime_format(self):
-                # to use df_expected to check the result
-                tm.assert_frame_equal(rs2, df_expected)
+            # since the reader returns a datetime object for dates, we need
+            # to use df_expected to check the result
+            tm.assert_frame_equal(rs2, df_expected)
 
-    def test_to_excel_interval_no_labels(self):
+    def test_to_excel_interval_no_labels(self, merge_cells, engine, ext):
         # GH19242 - test writing Interval without labels
-        _skip_if_no_xlrd()
-
-        with ensure_clean(self.ext) as path:
-            frame = DataFrame(np.random.randint(-10, 10, size=(20, 1)),
-                              dtype=np.int64)
-            expected = frame.copy()
-            frame['new'] = pd.cut(frame[0], 10)
-            expected['new'] = pd.cut(expected[0], 10).astype(str)
-            frame.to_excel(path, 'test1')
-            reader = ExcelFile(path)
-            recons = read_excel(reader, 'test1')
-            tm.assert_frame_equal(expected, recons)
-
-    def test_to_excel_interval_labels(self):
+        frame = DataFrame(np.random.randint(-10, 10, size=(20, 1)),
+                          dtype=np.int64)
+        expected = frame.copy()
+        frame['new'] = pd.cut(frame[0], 10)
+        expected['new'] = pd.cut(expected[0], 10).astype(str)
+        frame.to_excel(self.path, 'test1')
+        reader = ExcelFile(self.path)
+        recons = read_excel(reader, 'test1')
+        tm.assert_frame_equal(expected, recons)
+
+    def test_to_excel_interval_labels(self, merge_cells, engine, ext):
         # GH19242 - test writing Interval with labels
-        _skip_if_no_xlrd()
-
-        with ensure_clean(self.ext) as path:
-            frame = DataFrame(np.random.randint(-10, 10, size=(20, 1)),
-                              dtype=np.int64)
-            expected = frame.copy()
-            intervals = pd.cut(frame[0], 10, labels=['A', 'B', 'C', 'D', 'E',
-                                                     'F', 'G', 'H', 'I', 'J'])
-            frame['new'] = intervals
-            expected['new'] = pd.Series(list(intervals))
-            frame.to_excel(path, 'test1')
-            reader = ExcelFile(path)
-            recons = read_excel(reader, 'test1')
-            tm.assert_frame_equal(expected, recons)
-
-    def test_to_excel_timedelta(self):
+        frame = DataFrame(np.random.randint(-10, 10, size=(20, 1)),
+                          dtype=np.int64)
+        expected = frame.copy()
+        intervals = pd.cut(frame[0], 10, labels=['A', 'B', 'C', 'D', 'E',
+                                                 'F', 'G', 'H', 'I', 'J'])
+        frame['new'] = intervals
+        expected['new'] = pd.Series(list(intervals))
+        frame.to_excel(self.path, 'test1')
+        reader = ExcelFile(self.path)
+        recons = read_excel(reader, 'test1')
+        tm.assert_frame_equal(expected, recons)
+
+    def test_to_excel_timedelta(self, merge_cells, engine, ext):
```

Comment:
This failed without `xlwt` because the extension was hard-coded to `.xls` all the time, rather than using the `ext` that the fixture provides for all other test cases. I updated that to use the fixture, but in the process discovered that this doesn't work correctly with `openpyxl` so explicitly added a `pytest.xfail` and plan to open an issue to address separately